Mathlib/Data/List/GetD.lean

```diff
@@ -30,46 +30,48 @@ variable (d : α)
 theorem getD_eq_getElem {n : ℕ} (hn : n < l.length) : l.getD n d = l[n] := by
   grind
 
-theorem getD_map {n : ℕ} (f : α → β) : (map f l).getD n (f d) = f (l.getD n d) := by simp
+theorem getD_eq_getElem? (i : Fin l.length) : l.getD i d = l[i]?.get (by simp) := by
+  simp only [getD_eq_getElem?_getD, Fin.is_lt, getElem?_pos, Option.getD_some, Fin.getElem_fin,
+    Option.get_some]
+
+theorem getD_eq_get (i : Fin l.length) : l.getD i d = l.get i :=
+  getD_eq_getElem ..
+
+theorem getD_map {n : ℕ} (f : α → β) : (map f l).getD n (f d) = f (l.getD n d) := by
+  simp only [getD_eq_getElem?_getD, getElem?_map, Option.getD_map]
 
 theorem getD_eq_default {n : ℕ} (hn : l.length ≤ n) : l.getD n d = d := by
   grind
 
 theorem getD_reverse {l : List α} (i) (h : i < length l) :
     getD l.reverse i = getD l (l.length - 1 - i) := by
-  funext a
-  rwa [List.getD_eq_getElem?_getD, List.getElem?_reverse, ← List.getD_eq_getElem?_getD]
+  grind
 
 /-- An empty list can always be decidably checked for the presence of an element.
 Not an instance because it would clash with `DecidableEq α`. -/
-def decidableGetDNilNe (a : α) : DecidablePred fun i : ℕ => getD ([] : List α) i a ≠ a :=
+def decidableGetDNilNe (a : α) : DecidablePred fun i : ℕ => getD nil i a ≠ a :=
   fun _ => isFalse fun H => H getD_nil
 
 @[simp]
-theorem getElem?_getD_singleton_default_eq (n : ℕ) : [d][n]?.getD d = d := by cases n <;> simp
+theorem getElem?_getD_singleton_default_eq (n : ℕ) : [d][n]?.getD d = d := by
```

Comment:
Ideally this would be renamed with a deprecation to `getD_getElem_singleton_default`. But no need for that to happen here. :-)